scripts/intersections.py

```diff
@@ -0,0 +1,73 @@
+import configparser
+import os
+import warnings
+import geopandas as gpd
+import pandas as pd
+pd.options.mode.chained_assignment = None
+warnings.filterwarnings('ignore')
+
+CONFIG = configparser.ConfigParser()
+CONFIG.read(os.path.join(os.path.dirname(__file__), 'script_config.ini'))
+BASE_PATH = CONFIG['file_locations']['base_path']
+
+DATA_RAW = os.path.join(BASE_PATH, 'raw')
+DATA_PROCESSED = os.path.join(BASE_PATH, 'processed')
+DATA_RESULTS = os.path.join(BASE_PATH, 'results')
+
+path = os.path.join(DATA_RAW, 'countries.csv')
+
+# Population and floods layer
+population_folder = os.path.join(BASE_PATH, 'processed', 'KEN', 'population', 'shapefiles')
+flood_folder = os.path.join(BASE_PATH, 'processed', 'KEN', 'hazards', 'inunriver', 'shapefiles')
+
+# Output folder
+folder_out = os.path.join(DATA_RESULTS, 'KEN', 'pop_hazard')
+
+def intersect_layers(folder_1, folder_2):
+
+    for firstfiles in os.listdir(folder_1):
+        
+        try:
+
+            if firstfiles.endswith('.shp'):
+
+                first_shapefile = os.path.join(folder_1, firstfiles)
+                first_gdf = gpd.read_file(first_shapefile)
+
+                for secondfiles in os.listdir(folder_2):
+
+                    if secondfiles.endswith('.shp'):
+
+                        second_shapefile = os.path.join(folder_2, secondfiles)
+                        second_gdf = gpd.read_file(second_shapefile)
+
+                        if firstfiles == secondfiles:
+
+                            print('Intersecting layers for {}'.format(str(firstfiles).strip('.shp')))
+                            intersection = gpd.overlay(first_gdf, second_gdf, how = 'intersection')
+                            print(intersection)
```

Comment:
This print statement shouldn't really be here. Should be cleaned up prior to pushing, as this is really just here to help you debug.